harness/assert.js

```diff
@@ -101,6 +101,42 @@ assert.throws = function (expectedErrorConstructor, func, message) {
   throw new Test262Error(message);
 };
 
+assert.compareArray = function(actual, expected, message) {
+  message  = message === undefined ? '' : message;
+
+  if (typeof message === 'symbol') {
+    message = message.toString();
+  }
+
+  assert(actual != null, `Actual argument shouldn't be nullish. ${message}`);
+  assert(expected != null, `Expected argument shouldn't be nullish. ${message}`);
+  var format = compareArray.format;
+  var result = compareArray(actual, expected);
+
+  // The following prevents actual and expected from being iterated and evaluated
+  // more than once unless absolutely necessary.
+  if (!result) {
+    assert(false, `Actual ${format(actual)} and expected ${format(expected)} should have the same contents. ${message}`);
+  }
+};
+
+function compareArray(a, b) {
+  if (b.length !== a.length) {
+    return false;
+  }
+
+  for (var i = 0; i < a.length; i++) {
+    if (!assert._isSameValue(b[i], a[i])) {
+      return false;
+    }
+  }
+  return true;
+}
+
+compareArray.format = function(arrayLike) {
+  return `[${[].map.call(arrayLike, String).join(', ')}]`;
```

Comment:
```suggestion
  return `[${Array.prototype.join.call(arrayLike, ', ')}]`;
```
since join also calls toString?

Reply:
`[].map.call([Symbol()], String).join(', ') // 'Symbol()'`
`Array.prototype.join.call([Symbol()], ', ') // throws`

Reply:
ahhh right, because of symbols. fair point.

Reply:
That would break on symbols.
```js
Array.prototype.join.call(["foo", "bar"], ', ');
// => "foo, bar"
Array.prototype.join.call([Symbol("foo"), Symbol("bar")], ', ');
// => TypeError
```

Reply:
I'd withheld any implementation changes, but this is sufficiently minor that I pushed it and also a tighter validity check (non-primitive rather than non-nullish). But if any of that is too far, I'm happy to defer to a followup.